### ISSUE TITLE
[java] False negative: NonSerializableClass not reporting map and collection types

### ISSUE THREAD

**Issue by zbynek** [pmd/pmd#6547]
**Affects PMD Version:** 

7.22.0

**Rule:**

Please provide the rule name and a link to the rule documentation:
https://docs.pmd-code.org/latest/pmd_rules_java_errorprone.html#NonSerializableClass

**Description:**

Non-serializable fields are reported only if their type is not annotated with `@Serializable`, but not if they are collections/maps involving non-serializable types.

**Code Sample demonstrating the issue:**

```java
class NotSerializable{};
class SerializeMe implements Serializable {
   private List<NotSerializable> list;
   private Map<Integer, NotSerializable> map1;
   private Map<NotSerializable, Integer> map2;
}
```

**Expected outcome:**

PMD should report a violation at lines 3,4,5, but doesn't. This is a false-negative.

**Running PMD through:** *IDE*
